src/sinks/websocket/sink.rs

```diff
@@ -261,6 +261,11 @@ impl WebSocketSink {
 
         let bytes_sent = register!(BytesSent::from(Protocol("websocket".into())));
         let events_sent = register!(EventsSent::from(Output(None)));
+        let encode_as_binary = match self.encoder.serializer() {
+            codecs::encoding::Serializer::RawMessage(_) => true,
+            _ => false,
```

Comment:
Part of me thinks the codec itself should provide this information (whether it is returning a UTF-8 string or bytes). I'm ok to wait until we do this sort of thing in another place before we generalize though.

In addition to the `RawMessage` codec, data encoded by the `Native` and `Avro` codecs should also be passed along as binary.

Reply:
updated